cylc/flow/cfgspec/globalcfg.py

```diff
@@ -691,8 +728,19 @@ class GlobalConfig(ParsecConfig):
     _DEFAULT = None
     _HOME = os.getenv('HOME') or get_user_home()
     CONF_BASENAME = "global.cylc"
-    SITE_CONF_DIR = os.path.join(os.sep, 'etc', 'cylc', 'flow', CYLC_VERSION)
-    USER_CONF_DIR = os.path.join(_HOME, '.cylc', 'flow', CYLC_VERSION)
+
+    def __init__(self, *args, **kwargs):
+        self.SITE_CONF_PATH = (os.getenv('CYLC_SITE_CONF_PATH') or
+                               os.path.join(os.sep, 'etc', 'cylc', 'flow'))
+        self.USER_CONF_PATH = os.path.join(self._HOME, '.cylc', 'flow')
+        version_hierarchy = get_version_hierarchy(CYLC_VERSION)
+        self.CONF_DIR_HIERARCHY = [
+            *[(upgrader.SITE_CONFIG, os.path.join(self.SITE_CONF_PATH, ver))
+              for ver in version_hierarchy],
+            *[(upgrader.USER_CONFIG, os.path.join(self.USER_CONF_PATH, ver))
+              for ver in version_hierarchy]
+        ]
```

Comment:
If converting these class attributes to instance attributes make them lower case.

Reply:
It was only `CONF_DIR_HIERARCHY` that needed to be moved inside `__init__` because it's not possible to use another class attribute inside the list comprehension (https://stackoverflow.com/a/13913933/3217306). Seeing as this is the only reason for making it a instance attribute, would it make sense to leave it uppercase if the other two are changed to class attributes?